File: bookstore_frontend/src/services/OrderService.js
import {postRequest, postRequest_noReturn} from "../utils/ajax";

export const getOrdersByUserId = (userId, callback) => {
    const url = "/getOrdersByUserId";
    const data = {userId: userId};
    postRequest(url, data, callback);
}

export const getOrders = (callback) => {
    const url = "/getOrders";
    postRequest(url, null, callback);
}

export const saveOrderWithItems = (userId, address, itemIds, itemNums, district, zip, email) => {
    const url = "/saveOrderWithItems";
    const data = {
        userId: userId,
        address: address,
        itemIds: itemIds,
        itemNums: itemNums,
        district: district,
        zip: zip,
        email: email
    };
    postRequest_noReturn(url, data);
}

File: bookstore_backend/src/main/java/com/bookstore/service/OrderService.java
package com.bookstore.service;

import com.bookstore.entity.Order;
import java.sql.Timestamp;
import java.util.List;

public interface OrderService {
    List<Order> getOrdersByUserId(int userId);
    List<Order> getOrders();
    void saveOrderWithItems(int userId, String address, Timestamp datetime, List<Integer> itemIds, List<Integer> itemNums, String district, String zip, String email);
}
